src/schema/types/page.ts

```diff
@@ -20,45 +20,76 @@ export const PageType = new GraphQLObjectType({
         content: { type: GraphQLID },
         permissions: {
             type: AccessType,
-            resolve(parent, args, context) {
+            async resolve(parent, args, context) {
                 const ability: AppAbility = context.user.ability;
-                return ability.can('update', parent) ? parent.permissions : null;
+                if (ability.can('update', parent)) {
+                    return parent.permissions;
+                } else {
+                    const application = await Application.findOne(Application.accessibleBy(ability, 'read').where({ pages: parent._id }).getFilter(), 'id');
+                    if (application) {
+                        return parent.permissions;
+                    }
+                }
+                return null;
             }
         },
         application: {
             type: ApplicationType,
             resolve(parent, args, context) {
                 const ability: AppAbility = context.user.ability;
-                return Application.findOne( { pages: parent.id } ).accessibleBy(ability, 'read');
+                return Application.findOne(Application.accessibleBy(ability, 'read').where({ pages: parent._id }).getFilter());
             }
         },
-        // TODO: fix all of that
         canSee: {
             type: GraphQLBoolean,
             async resolve(parent, args, context) {
                 const ability: AppAbility = context.user.ability;
-                return ability.can('read', new Page(parent));
+                if (ability.can('read', parent)) {
+                    return true;
+                } else if (context.user.isAdmin) {
+                    const application = await Application.findOne(Application.accessibleBy(ability, 'read').where({ pages: parent._id }).getFilter());
+                    return !!application;
+                }
+                return false;
             }
         },
         canCreate: {
             type: GraphQLBoolean,
-            resolve(parent, args, context) {
+            async resolve(parent, args, context) {
```

Comment:
@PacomeRivier can I remove this one ?

Reply:
Yes it should be removed in another ticket anyway !